### PR TITLE
Simplifying DNS resolution

### PR DESCRIPTION
**Pull request by 3alpha**
When dappnode/DNP_DAPPMANAGER#625 gets resolved, there won't be any need for full BIND server, just dnscrypt-proxy for outside access. Name resolution will be done by Docker's internal DNS and that will ensure more resiliency because internal resolution will not depend on this package and only purpose of this package will be in forwarding DNS requests by users connected to the VPN and Wifi.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
       driver: journald
+
 networks:
-  network:
-    driver: bridge
-    ipam:
-      config:
-        - subnet: 172.33.0.0/16
+  dncore_network:
+    external: true
```
Comment (dapplion) [inline docker-compose.yml:16]: We were debating some time ago about how to handle the dncore_network. By declaring it as external you are expecting that this package will be started strictly after another that creates the network (i.e. the DAPPMANAGER). What do you think? Should we change the dncore_network setup in another PR?

File: build/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    fi
+
+    if [ -z $INTERNAL_IP ] || [ -z $DOMAIN ]; then
+        echo "Retrying in one second..."
+        sleep 1
+    else
+        echo "Done, found $DOMAIN, $INTERNAL_IP"
+        break
+    fi
+done
```
Comment (dapplion) [inline build/entrypoint.sh:21]: While this loop is going what would actions could be affected? Is it possible to start dnscrypt immediately and reboot it latter when the `$DOMAIN, $INTERNAL_IP` is available?

File: build/dnscrypt-proxy.toml [hunk 3]
```diff
@@ -235,7 +235,7 @@ block_ipv6 = false
 ## example.com 9.9.9.9
 ## example.net 9.9.9.9,8.8.8.8,1.1.1.1
 
-# forwarding_rules = 'forwarding-rules.txt'
+forwarding_rules = 'forwarding-rules.txt'
```
Comment (dapplion) [inline build/dnscrypt-proxy.toml:238]: Could you add a comment above explaining the reasoning of the contents of `forwarding-rules.txt`? Same for `cloaking-rules.txt`

File: docker-compose.yml (hunk 1 above)
Comment (3alpha) [reply docker-compose.yml:16]: This is the main reason PR is still a draft. That change is just for development purposes so when I compose up it joins proper network.

File: build/dnscrypt-proxy.toml (hunk 3 above)
Comment (3alpha) [reply build/dnscrypt-proxy.toml:238]: Forwarding rules make sure that .dappnode and .eth domains get resolved by Docker's internal DNS. Cloaking rules make sure that we override resolution of xxxxxx.dydns.dappnode.io from public to private IP.

File: build/entrypoint.sh (hunk 2 above)
Comment (3alpha) [reply build/entrypoint.sh:21]: This is only necessary for cloaking rules (overriding xxxxxx.dydns.dappnode.io resolution) and is not critical for general functionality of resolver. I am not sure whether dns-crypt even needs to be restarted when updating these files. But if it has to be restarted, we would have to figure out way to detect when that becomes available and restart it. But, I don't that would be particularly difficult.

**Comment by pablomendezroyo** [comment]
@3alpha Is this PR required anymore?

**Comment by 3alpha** [comment]
This has been done.

**Pull request closed without merging**